src/coreclr/tools/aot/ILCompiler/Program.cs

```diff
@@ -432,13 +436,18 @@ private int Run(string[] args)
             InstructionSetSupportBuilder instructionSetSupportBuilder = new InstructionSetSupportBuilder(_targetArchitecture);
 
             // The runtime expects certain baselines that the codegen can assume as well.
-            if ((_targetArchitecture == TargetArchitecture.X86) || (_targetArchitecture == TargetArchitecture.X64))
+            if (_targetArchitecture == TargetArchitecture.ARM64 && _targetOS == TargetOS.OSX)
             {
-                instructionSetSupportBuilder.AddSupportedInstructionSet("sse2"); // Lower baselines included by implication
+                // For osx-arm64 we know that apple-m1 is a baseline
+                bool added = instructionSetSupportBuilder.AddSupportedInstructionSet("apple-m1");
+                Debug.Assert(added);
             }
-            else if (_targetArchitecture == TargetArchitecture.ARM64)
+            else if ((_targetArchitecture == TargetArchitecture.X86) ||
```

Comment:
We should require `generic` to be defined for all architectures, so that this condition is not required.

Or drop `generic` as recognized name since it is the default. There is no real reason for anybody to use it.

Reply:
@jkotas I was thinking of if we decide to bump the baseline for crossgen2 to avx2, generic will be a quick ability to override that behavior, e.g. `clang -mcpu=generic` allows me to limit my binary to arm8.0 on my Apple M1.

Reply:
> clang -mcpu=generic allows me to limit my binary to arm8.0 on my Apple M1.

I do not think that what's currently implemented in this PR allows you to override or subtract from the default like this.